api/etcd/rulesets_test.go

```diff
@@ -55,12 +56,12 @@ func createRuleset(t *testing.T, s *RulesetService, path string, rules ...*rule.
 	}
 
 	rs, err := s.Get(context.Background(), path, "")
-	require.NoError(t, err)
+	assert.NoError(t, err)
 	return rs
 }
 
 func createBoolRuleset(t *testing.T, s *RulesetService, path string, rules ...*rule.Rule) *regula.Ruleset {
 	err := s.Create(context.Background(), path, &regula.Signature{ReturnType: "bool"})
-	require.False(t, err != nil && err != api.ErrAlreadyExists)
+	assert.False(t, err != nil && err != api.ErrAlreadyExists)
```

Comment:
This function is a helper which means that it can be called anywhere, including in a goroutine.
Panic'ing in a goroutine (that's what require does) doesn't let you run the `defer` functions that are not in the same goroutine which can cause problems. Also, error messages when test fail are obscure. With `assert`, it's all fine, so I'm more for using assert in helpers and require in tests, though that's not a hard rule.

Reply:
Is it worth adding `t.Helper()` call in it?

Reply:
Good idea, I'll add it

Reply:
I have to say I think I agree with @YassineE here. If a testing helper takes a `*testing.T` argument and doesn't return an indication of success, I think it's reasonable to have it fail immediately, so the test doesn't proceed with partially wrong/confusing data. Perhaps a name like `requireCreateBoolRuleset` might provide more indication that it fails immediately on error?